### PR TITLE
Added oauth2_auth_token and per_rpc_creds Node interop tests

### PR DESCRIPTION
**Pull request by murgatroid99**

### CODE REVIEW

File: src/node/interop/interop_client.js [hunk 1]
```diff
@@ -318,6 +318,58 @@ function authTest(expected_user, scope, client, done) {
   });
 }
 
+function oauth2Test(expected_user, scope, per_rpc, client, done) {
+  (new GoogleAuth()).getApplicationDefault(function(err, credential) {
+    assert.ifError(err);
+    var arg = {
+      response_type: 'COMPRESSABLE',
```
Comment (jtattermusch) [inline src/node/interop/interop_client.js]: The spec only mentions fill_username and fill_oauth_scope for the new interop tests, but I think that alright.

Comment (murgatroid99) [inline src/node/interop/interop_client.js]: Oh, I didn't even notice. I just copied from the service account test. I'll delete that stuff to conform to the spec.

**Comment by jtattermusch** [comment]
LGTM, feel free to merge once Jenkins is green.

**Comment by murgatroid99** [comment]
Jenkins is green enough: the Node tests are passing.

**Pull request merged**